[folly] build failure

**Host Environment**
- OS: Windows
- Compiler: vs2017

**To Reproduce**
Steps to reproduce the behavior:
.\vcpkg\vcpkg install folly:x64-windows-static

**Failure logs** 
vcpkg\buildtrees\folly\x64-windows-dbg\CMakeFiles\CMakeTmp\CheckIncludeFile.c(1): fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory
ninja: build stopped: subcommand failed.
vcpkg\buildtrees\folly\x64-windows-dbg\CMakeFiles\CMakeTmp\CheckSymbolExists.cxx(2): fatal error C1083: Cannot open include file: 'ucontext.h': No such file or directory
ninja: build stopped: subcommand failed.

Thanks for your reply. There are message in config-x64-windows-static-out.log:
```
CMake Error at CMake/FollyFunctions.cmake:95 (if):
  if given arguments:

    "C:/code/C++/vcpkg/buildtrees/folly/src/1.06.14.00-b189899f14.clean/folly/Benchmark.cpp" "MATCHES" "^C:/code/C++/vcpkg/buildtrees/folly/src/1.06.14.00-b189899f14.clean/folly/build/"

  Regular expression
  "^C:/code/C++/vcpkg/buildtrees/folly/src/1.06.14.00-b189899f14.clean/folly/build/"
  cannot compile
Call Stack (most recent call first):
  CMakeLists.txt:149 (REMOVE_MATCHES_FROM_LISTS)


-- Configuring incomplete, errors occurred!
```
The reason is your _VCPKG_ROOT_ path have **++** in _C:/code/C++/vcpkg_, you can create a new directory which doesn't contain character. I will confirm if this issue is a bug or by design soon.

~~~
CMake Error at CMake/FollyFunctions.cmake:95 (if):
  if given arguments:

    "C:/code/C++/vcpkg/buildtrees/folly/src/1.06.14.00-b189899f14.clean/folly/Benchmark.cpp" "MATCHES" "^C:/code/C++/vcpkg/buildtrees/folly/src/1.06.14.00-b189899f14.clean/folly/build/"

  Regular expression
  "^C:/code/C++/vcpkg/buildtrees/folly/src/1.06.14.00-b189899f14.clean/folly/build/"
  cannot compile
Call Stack (most recent call first):
  CMakeLists.txt:149 (REMOVE_MATCHES_FROM_LISTS)


-- Configuring incomplete, errors occurred!
See also "C:/code/C++/vcpkg/buildtrees/folly/x64-windows-static-dbg/CMakeFiles/CMakeOutput.log".
See also "C:/code/C++/vcpkg/buildtrees/folly/x64-windows-static-dbg/CMakeFiles/CMakeError.log".
RegularExpression::compile(): Nested *?+.
RegularExpression::compile(): Error in compile.
~~~

Don't use `+` in your directory names. Folly stumbles over `C++` because it uses a raw filesystem paths as regular expressions. `+` has a special meaning.

--- a/ports/folly/fix-character-in-folder.patch
+++ b/ports/folly/fix-character-in-folder.patch
@@ -1,0 +1,41 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c58dedc..c9561c1 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -113,6 +113,8 @@ set(
+   FOLLY_DIR_PREFIXES
+   "${CMAKE_CURRENT_SOURCE_DIR}:${CMAKE_CURRENT_BINARY_DIR}"
+ )
++# https://gitlab.kitware.com/cmake/cmake/-/issues/18580#note_1405108
++string(REGEX REPLACE "(.)" "\\\\\\1" FOLLY_DIR_REGEX_ESCAPED "${FOLLY_DIR}")
+ 
+ include(GNUInstallDirs)
+ 
+@@ -163,12 +165,12 @@ auto_sources(hfiles "*.h" "RECURSE" "${FOLLY_DIR}")
+ # library sources.  Test sources are listed separately below.
+ REMOVE_MATCHES_FROM_LISTS(files hfiles
+   MATCHES
+-    "^${FOLLY_DIR}/build/"
+-    "^${FOLLY_DIR}/docs/examples/"
+-    "^${FOLLY_DIR}/logging/example/"
+-    "^${FOLLY_DIR}/(.*/)?test/"
+-    "^${FOLLY_DIR}/(.*/)?tool/"
+-    "^${FOLLY_DIR}/facebook/"
++    "^${FOLLY_DIR_REGEX_ESCAPED}/build/"
++    "^${FOLLY_DIR_REGEX_ESCAPED}/docs/examples/"
++    "^${FOLLY_DIR_REGEX_ESCAPED}/logging/example/"
++    "^${FOLLY_DIR_REGEX_ESCAPED}/(.*/)?test/"
++    "^${FOLLY_DIR_REGEX_ESCAPED}/(.*/)?tool/"
++    "^${FOLLY_DIR_REGEX_ESCAPED}/facebook/"
+     "Benchmark.cpp$"
+     "Test.cpp$"
+ )
+@@ -177,7 +179,7 @@ REMOVE_MATCHES_FROM_LISTS(files hfiles
+ if (${FOLLY_NO_EXCEPTION_TRACER})
+   REMOVE_MATCHES_FROM_LISTS(files hfiles
+     MATCHES
+-      "^${FOLLY_DIR}/debugging/exception_tracer/"
++      "^${FOLLY_DIR_REGEX_ESCAPED}/debugging/exception_tracer/"
+   )
+ endif()
+ 

--- a/ports/folly/portfile.cmake
+++ b/ports/folly/portfile.cmake
@@ -18,6 +18,7 @@ vcpkg_from_github(
         disable-uninitialized-resize-on-new-stl.patch
         fix-unistd-include.patch
         fix-fmt11-cmake.patch
+        fix-character-in-folder.patch # https://github.com/facebook/folly/commit/0f698c382d34a2e139d5b2c071151ab33e1ffbd3
 )
 
 file(REMOVE "${SOURCE_PATH}/CMake/FindFmt.cmake")

--- a/ports/folly/vcpkg.json
+++ b/ports/folly/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "folly",
   "version-string": "2024.08.26.00",
+  "port-version": 1,
   "description": "An open-source C++ library developed and used at Facebook. The library is UNSTABLE on Windows",
   "homepage": "https://github.com/facebook/folly",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2826,7 +2826,7 @@
     },
     "folly": {
       "baseline": "2024.08.26.00",
-      "port-version": 0
+      "port-version": 1
     },
     "font-chef": {
       "baseline": "1.1.0",

--- a/versions/f-/folly.json
+++ b/versions/f-/folly.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "3853e408c3b7432b3db7080a38252eb07cbe31d8",
+      "version-string": "2024.08.26.00",
+      "port-version": 1
+    },
+    {
       "git-tree": "fc8cca13cb39a5e46abdeaedbd23156c5260ff47",
       "version-string": "2024.08.26.00",
       "port-version": 0

